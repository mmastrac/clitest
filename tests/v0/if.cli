#!/usr/bin/env clitest --v0

if true {
    # This should print
    $ printf "true\n"
    *
}

if false {
    # This should not print
    $ printf "false\n" && exit 1
    *
}

set x 1;
set y 1;

if x == 1 {
    # This should print
    $ printf "x is 1\n"
    *
}

if x == 2 {
    # This should not print
    $ printf "x is 2\n" && exit 1
    *
}

if x != 2 {
    # This should print
    $ printf "x is not 2\n"
    *
}

if x == $y {
    $ printf "x is equal to y\n"
    *
}
