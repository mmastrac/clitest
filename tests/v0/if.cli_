#!/usr/bin/env clitest --v0

if true {
    # This should print
    $ echo "true"
    *
}

if false {
    # This should not print
    $ echo "false" && exit 1
    *
}
