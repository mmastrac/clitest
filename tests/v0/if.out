Running <root>/if.cli ...

printf "true\n"
---
true
---
✅ OK

set x 1

set y 1

printf "x is 1\n"
---
x is 1
---
✅ OK

printf "x is not 2\n"
---
x is not 2
---
✅ OK

printf "x is equal to y\n"
---
x is equal to y
---
✅ OK

<root>/if.cli PASSED
